src/liboslexec/llvm_instance.cpp

```diff
@@ -638,6 +638,99 @@ BackendLLVM::llvm_assign_initial_value(const Symbol& sym, bool force)
             // source didn't have them.
             if (sym.has_derivs() && !symloc->derivs)
                 ll.op_memset(ll.offset_ptr(dstptr, size), 0, 2 * size);
+        } else if (renderer()->supports("build_interpolated_getter")) {
+            InterpolatedGetterSpec spec;
+            renderer()->build_interpolated_getter(group(), symname, type,
+                                                  sym.has_derivs(), spec);
+            if (!spec.function_name().empty()) {
+                std::vector<llvm::Value*> args;
+                args.reserve(spec.arg_count() + 1);
+                for (size_t index = 0; index < spec.arg_count(); index++) {
+                    const auto& arg = spec.arg(index);
```

Comment:
Even though the code flow says it, maybe add a comment here explaining that the argument could be dynamic value identified by an enum or constant value that can directly be emitted.

Reply:
I'd rather keep the virtual functions in a good logical order when reading the header. Appending to the end is a bit of a suspect way to avoiding ABI break, I don't think it's guaranteed.

We're allowed to break ABI compatibility in main. We just need to adjust the version numbers to reflect it. Which we can do in a subsequent commit (we already need a subsequent commit in order to update the reference commit for ABI compatibility, so this is not a problem).

Reply:
I think I put this comment in the wrong place!